lib/cufft/fft.jl

```diff
@@ -8,9 +8,21 @@ import AbstractFFTs: plan_fft, plan_fft!, plan_bfft, plan_bfft!, plan_ifft,
 
 using LinearAlgebra
 
-Base.:(*)(p::Plan{T}, x::DenseCuArray) where {T} = p * copy1(T, x)
-Base.:(*)(p::ScaledPlan, x::DenseCuArray) = rmul!(p.p * x, p.scale)
+Base.:(*)(p::Plan{T}, x::StridedCuArray) where {T} = p * copy1(T, x)
+Base.:(*)(p::ScaledPlan, x::StridedCuArray) = rmul!(p.p * x, p.scale)
 
+# FakeCuArray is used to pass the strides and size information into create_plan,
```

Comment:
I still don't understand why this is required. FFTW.jl does so because it passes the fake array to FFTW, represented as a null pointer. But here we don't actually pass these fake CuArrays to CUFFT, so why not just pass the strides and element type instead?

Reply:
I use it mainly because I refered the code from `FFTW.jl`, and `creat_plan(::Type,X::AbstractArray,Y::AbstractArray)` seems intuitive.

On the other hand, we have to check the CUDA version during the runtime. [Line 212, 213]
If we only have no type information, then we have check the layout via size and strides.
（Although this should not be a performance bottleneck）